Smart Village App/views/calendar/common/_settings_participation.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+use humhub\components\View;
+use humhub\modules\calendar\models\forms\CalendarEntryForm;
+use humhub\modules\calendar\models\participation\ParticipationSettings;
+use humhub\widgets\Button;
+
+/* @var $this View */
+/* @var $participationSettings ParticipationSettings */
+
+$helpBlock = $participationSettings->isGlobal()
+    ? Yii::t('CalendarModule.config', 'Here you can configure default settings for new calendar events. These settings can be overwritten on space/profile level.')
+    : Yii::t('CalendarModule.config', 'Here you can configure default settings for new calendar events.') ;
+
+?>
+
+<div class="panel-body" data-ui-widget="calendar.Form">
+    <h4>
+        <?= Yii::t('CalendarModule.config', 'Default participation settings'); ?>
+        <?php if ($participationSettings->showResetButton()) : ?>
+            <?= Button::defaultType(Yii::t('CalendarModule.config', 'Reset'))
+                ->action('client.pjax.post', $participationSettings->getResetButtonUrl())->link()->right()->sm()?>
+        <?php endif; ?>
+    </h4>
+
+    <div class="help-block">
+        <?= $helpBlock ?>
+    </div>
+
+    <?= $form->field($participationSettings, 'participation_mode')->dropDownList(CalendarEntryForm::getParticipationModeItems(), ['data-action-change' => 'changeParticipationMode']) ?>
+</div>
```

Comment:
please remember to always provide the original file contents that gets overridden as we otherwise cannot compare with anything and review changes. thanks

Reply:
> please remember to always provide the original file contents that gets overridden as we otherwise cannot compare with anything and review changes. thanks

Okay @donni106 
The original file of _setting_participation.php 

<?php

use humhub\components\View;
use humhub\modules\calendar\models\forms\CalendarEntryForm;
use humhub\modules\calendar\models\participation\ParticipationSettings;
use humhub\widgets\Button;

/* @var $this View */
/* @var $participationSettings ParticipationSettings */

$helpBlock = $participationSettings->isGlobal()
    ? Yii::t('CalendarModule.config', 'Here you can configure default settings for new calendar events. These settings can be overwritten on space/profile level.')
    : Yii::t('CalendarModule.config', 'Here you can configure default settings for new calendar events.') ;

?>

<div class="panel-body" data-ui-widget="calendar.Form">
    <h4>
        <?= Yii::t('CalendarModule.config', 'Default participation settings'); ?>
        <?php if ($participationSettings->showResetButton()) : ?>
            <?= Button::defaultType(Yii::t('CalendarModule.config', 'Reset'))
                ->action('client.pjax.post', $participationSettings->getResetButtonUrl())->link()->right()->sm()?>
        <?php endif; ?>
    </h4>

    <div class="help-block">
        <?= $helpBlock ?>
    </div>

    <?= $form->field($participationSettings, 'participation_mode')->dropDownList(CalendarEntryForm::getParticipationModeItems(), ['data-action-change' => 'changeParticipationMode']) ?>
    <div class="participationOnly" style="<?= $participationSettings->isParticipationAllowed() ? '' : 'display:none' ?>">
        <?= $form->field($participationSettings, 'allow_decline')->checkbox() ?>
        <?= $form->field($participationSettings, 'allow_maybe')->checkbox() ?>
    </div>
</div>

**We only remove these two checkboxes in the file**

Reply:
ok thanks. i have to say that this is very hard to read:

![Bildschirmfoto 2022-07-13 um 10 17 22](https://user-images.githubusercontent.com/1942953/178685496-de99106c-2676-4b03-9325-6382921c49f5.png)

there is unformatted code, bold unformatted code and formatted code. what does that mean?
is there maybe a link we can have in the future to the humhub codebase where the original files are located and be useful for review, instead of copying parts of it? let's try that in future PRs or what do you think could be the best way for a good comparison?

Reply:
Hi @donni106 
By mistake i did not put original file content in code block that's why code is showing unformatted. I apologize for that and for all the theme related point we are using humhub 1.9.3 version (your version).

Reply:
`<?php

use humhub\components\View;
use humhub\modules\calendar\models\forms\CalendarEntryForm;
use humhub\modules\calendar\models\participation\ParticipationSettings;
use humhub\widgets\Button;

/* @var $this View */
/* @var $participationSettings ParticipationSettings */

$helpBlock = $participationSettings->isGlobal()
    ? Yii::t('CalendarModule.config', 'Here you can configure default settings for new calendar events. These settings can be overwritten on space/profile level.')
    : Yii::t('CalendarModule.config', 'Here you can configure default settings for new calendar events.') ;

?>

<div class="panel-body" data-ui-widget="calendar.Form">
    <h4>
        <?= Yii::t('CalendarModule.config', 'Default participation settings'); ?>
        <?php if ($participationSettings->showResetButton()) : ?>
            <?= Button::defaultType(Yii::t('CalendarModule.config', 'Reset'))
                ->action('client.pjax.post', $participationSettings->getResetButtonUrl())->link()->right()->sm()?>
        <?php endif; ?>
    </h4>

    <div class="help-block">
        <?= $helpBlock ?>
    </div>

    <?= $form->field($participationSettings, 'participation_mode')->dropDownList(CalendarEntryForm::getParticipationModeItems(), ['data-action-change' => 'changeParticipationMode']) ?>
    <div class="participationOnly" style="<?= $participationSettings->isParticipationAllowed() ? '' : 'display:none' ?>">
        <?= $form->field($participationSettings, 'allow_decline')->checkbox() ?>
        <?= $form->field($participationSettings, 'allow_maybe')->checkbox() ?>
    </div>
</div>
`

Reply:
![image](https://user-images.githubusercontent.com/103625795/178714212-a38e8287-69c0-4d96-b54b-52677f06ae24.png)

This bold code nothing but this is reset button code.

![image](https://user-images.githubusercontent.com/103625795/178714297-f1d81579-64d4-4e4a-a4ba-9ec6bd003f58.png)

Thanks!

Reply:
Hi @donni106 
Okay, So for original file reference we have three options:

1. First, go to humhub github directory with same version(1.9.3) you installed and attached the url of that file you want to compare (the orginal file) like

**So, branch we select is v1.9 (refer to humhub version 1.9)**
https://github.com/humhub/humhub/blob/v1.9/protected/humhub/modules/stream/widgets/views/wallStreamFilterNavigation.php

We are referring the latest commit of that file

![image](https://user-images.githubusercontent.com/103625795/178911147-1513bd10-b811-44f8-8690-bb41238356f6.png)

2. Paste the file code in gist.github and share the file with url.

3. Attached the file (in zip) here(github)

What will be your suggestion? feel free to share with us 
Thanks!

Reply:
all right, the original links to files should then be always part of the PR description.
i don't think that we need gist and zip. we could directly link to the files. as they are in git, there is always a direct reference to a file in time.
so as the first try, please link to the original calendar _settings_participation.php file here.

Reply:
that is perfect this way! thanks